### PR TITLE
When a non-numeric option is provided, the game gets stuck in an infinite loop

### PR DESCRIPTION
**Pull request by haroon**
`cin` goes in error state which needs to be cleared before proceeding. To overcome this, following needs to be done:
Test for error state.
Clear the error state.
Flush it out and prompt the user for valid input.

### CODE REVIEW

File: Turn/src/Game.cpp [hunk 1]
```diff
@@ -340,10 +343,21 @@ void Game::HowToPlay() {
...
 	}
 }
+
+int Game::GetChoice()
+{
+	int choice = -1;cin >> choice;
+	while (!(cin >> choice)) {
+		cin.clear();
+		cin.ignore(numeric_limits<streamsize>::max(), '\n');
+		cout << "Invalid input.  Please try again: ";
```
Comment (tagniam) [inline Turn/src/Game.cpp]: I would prefer calling `Sleep(SLEEP_MS)` after the error message, then redraw the main menu. This way the screen won't be littered with invalid input messages if they enter a lot of incorrect keys.

**Comment by haroon** [comment]
Updated. Also combined prompts at one place for easy management.

**Comment by haroon** [comment]
I think it would be better to separate user input from the game loop. The game loop itself will be cleaner and the prompt/input validation can be placed separately.

**Comment by tagniam** [comment]
Ok that sounds alright to me. Thanks for the contribution @haroon!

**Pull request merged**